dev/compare_models.py

```diff
@@ -0,0 +1,307 @@
+        #results_df = pd.concat([results_df, temp_df])
+
+        if(args.run_test):
+            for config in config_list:
+                # Delete path_pred
+                path_pred = os.path.join(config['log_directory'], 'pred_masks')
+                if os.path.isdir(path_pred) and n_iterations > 1:
+                    try:
+                        shutil.rmtree(path_pred)
+                    except OSError as e:
+                        print("Error: %s - %s." % (e.filename, e.strerror))
+
+            test_scores = pool.map(test_worker, config_list)
+            test_df = pd.DataFrame(test_scores, columns=[
+                                   'log_directory', 'test_dice'])
+            combined_df = val_df.set_index('log_directory').join(
+                test_df.set_index('log_directory'))
+            combined_df = combined_df.reset_index()
+
+        else:
+            combined_df = val_df
+
+        results_df = pd.concat([results_df, combined_df])
+
+    # Compute avg, std, p-values
+    if(n_iterations > 1):
+        num_configs = len(config_list)
+        avg = results_df.groupby(['log_directory']).mean()
+        std = results_df.groupby(['log_directory']).std()
+
+        print("Average dataframe")
+        print(avg)
+        print("Standard deviation dataframe")
+        print(std)
+
+        p_values = np.zeros((num_configs, num_configs))
+        i, j = 0, 0
+        for confA in config_list:
+            print(confA["log_directory"])
+            j = 0
+            for confB in config_list:
+                if args.run_test:
+                    p_values[i, j] = ttest_ind_from_stats(mean1=avg.loc[confA["log_directory"]]["test_dice"], std1=std.loc[confA["log_directory"]]["test_dice"],
+                                                          nobs1=n_iterations, mean2=avg.loc[confB["log_directory"]]["test_dice"], std2=std.loc[confB["log_directory"]]["test_dice"], nobs2=n_iterations).pvalue
+                else:
+                    p_values[i, j] = ttest_ind_from_stats(mean1=avg.loc[confA["log_directory"]]["best_validation_dice"], std1=std.loc[confA["log_directory"]]["best_validation_dice"],
+                                                          nobs1=n_iterations, mean2=avg.loc[confB["log_directory"]]["best_validation_dice"], std2=std.loc[confB["log_directory"]]["best_validation_dice"], nobs2=n_iterations).pvalue
+                j += 1
+            i += 1
+
+        print("P-values array")
+        print(p_values)
+
+    # Merge config and results in a df
+    config_df = pd.DataFrame.from_dict(config_list)
+    keep = list(param_dict.keys())
+    keep.append("log_directory")
+    config_df = config_df[keep]
+
+    results_df = config_df.set_index('log_directory').join(results_df.set_index('log_directory'))
+    results_df = results_df.sort_values(by=['best_validation_loss'])
+
+    results_df.to_pickle("output_df.pkl")
```

Comment:
can we avoid pkl output and prefer csv files which are human readable? (and could be opened by many more software)